server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTreeBelowSG.java

```diff
@@ -177,14 +169,28 @@ public void init() throws IOException {
     File mtreeSnapshot = SystemFileFactory.INSTANCE.getFile(mtreeSnapshotPath);
     long time = System.currentTimeMillis();
     if (mtreeSnapshot.exists()) {
-      this.root = deserializeFrom(mtreeSnapshot).root;
-      logger.debug(
-          "spend {} ms to deserialize mtree from snapshot", System.currentTimeMillis() - time);
+      IStorageGroupMNode recoveredTree =
+          deserializeFrom(mtreeSnapshot, storageGroupMNode.getFullPath());
+      if (recoveredTree != null) {
+        this.storageGroupMNode.getParent().replaceChild(storageGroupMNode.getName(), recoveredTree);
```

Comment:
From this line, I cannot infer that some attributes inside recoveredTree has been modified. But it actually did. 

Suggest to use more clear way to implement the changes here...

Reply:
Some comments have been added. Hope it will help understand.

Reply:
LOL